src/geometry_base/kd_tree_strategy.cc

```diff
@@ -0,0 +1,143 @@
+
+#include <cstdlib>
+
+#include "maliput/math/kd_tree.h"
+
+namespace maliput {
+namespace geometry_base {
+
+api::RoadPositionResult KDTreeStrategy::DoToRoadPosition(const api::InertialPosition& inertial_position,
+                                                         const std::optional<api::RoadPosition>& hint) const {
+  if (hint.has_value()) {
+    MALIPUT_THROW_UNLESS(hint->lane != nullptr);
+    const maliput::api::LanePositionResult lane_pos = hint->lane->ToLanePosition(inertial_position);
+    return {{hint->lane, lane_pos.lane_position}, lane_pos.nearest_position, lane_pos.distance};
+  }
+  const auto lane_id = this->ClosestLane(inertial_position.xyz());
+  const auto lane = rg_->ById().GetLane(lane_id);
+  const auto lane_position_result = lane->ToLanePosition(inertial_position);
+  return {
+      {lane, lane_position_result.lane_position}, lane_position_result.nearest_position, lane_position_result.distance};
+}
+
+std::vector<api::RoadPositionResult> KDTreeStrategy::DoFindRoadPositions(const api::InertialPosition& inertial_position,
+                                                                         double radius) const {
+  const auto lane_ids = this->ClosestLanes(inertial_position.xyz(), radius);
+  std::vector<api::RoadPositionResult> road_positions;
+  for (const auto& current_lane : lane_ids) {
+    const auto lane = rg_->ById().GetLane(current_lane);
+    const auto lane_position = lane->ToLanePosition(inertial_position);
+    if (lane_position.distance <= radius) {
+      api::RoadPositionResult road_position{
+          {lane, lane_position.lane_position}, lane_position.nearest_position, lane_position.distance};
+      road_positions.push_back(road_position);
+    }
+  }
+  return road_positions;
+}
+
+void KDTreeStrategy::Init() {
+  const auto lanes = rg_->ById().GetLanes();
+
+  std::deque<maliput::geometry_base::MaliputPoint> points;
+  for (const auto& lane : lanes) {
+    const auto lane_length = lane.second->length();
+    for (double s = 0; s <= lane_length; s += 0.1) {
+      const auto lane_bounds = lane.second->lane_bounds(s);
+      for (double r = lane_bounds.min(); r <= lane_bounds.max(); r += 0.1) {
+        const auto inertial_pos = lane.second->ToInertialPosition({s, r, 0. /* h */}).xyz();
+        const maliput::geometry_base::MaliputPoint point{{inertial_pos.x(), inertial_pos.y(), inertial_pos.z()},
+                                                         lane.first};
+        points.push_back(point);
+      }
+    }
+  }
+  kd_tree_ = (std::make_unique<math::KDTree3D<MaliputPoint>>(std::move(points)));
+}
+
+maliput::api::LaneId KDTreeStrategy::do_closest_lane(const maliput::math::Vector3& point) const {
+  // Given that this method should actually obtain the lanes within a tolerance radius. Afterwards, the lane
+  // could be obtained using the point being closer to the centerline.
+  const api::InertialPosition& inertial_position{point.x(), point.y(), point.z()};
```

Comment:
Why not receive the InertialPosition directly and get the 3D point inside the function if required?

Reply:
This method was deleted on some recent changes and no longer creates an additional InertialPosition.